Error reporting for incorrect yaml syntax too cryptic

### Description

* Autosubmit version: 4.1.14.
* Machine/VM/environment name (if applicable): climatedt vm
* Experiment ID (if applicable): o010 and o00a
* Experiment tasks & log path if applicable):
  * e.g. `/appl/AS/AUTOSUBMIT_DATA/o010/tmp/ASLOGS/20251013_151319_run.log`

I had a typo (missing ') in my main.yml and the error message was not helpful at all. 
```
[ERROR] 2025-10-13 15:19:38,320 Arguments provided: Namespace(version=False, logfile='DEBUG', logconsole='WARNING', command='run', expid='o010', notransitive=False, update_version=False, start_time=None, start_after=None, run_only_members=None, profile=False)
[ERROR] 2025-10-13 15:19:38,321 This is the experiment: <o010> which had an issue with the command: <run> and it is currently using the Autosubmit Version: 4.1.14.
2025-10-13 15:19:38,321 Traceback (most recent call last):
  File "/appl/AS/4.1.14/lib64/python3.9/site-packages/autosubmit/scripts/autosubmit.py", line 105, in main
    return_value = Autosubmit.run_command(args)
  File "/appl/AS/4.1.14/lib64/python3.9/site-packages/autosubmit/autosubmit.py", line 726, in run_command
    return Autosubmit.run_experiment(args.expid, args.notransitive,args.start_time,args.start_after, args.run_only_members, args.profile)
  File "/appl/AS/4.1.14/lib64/python3.9/site-packages/autosubmit/autosubmit.py", line 2205, in run_experiment
    job.update_parameters(as_conf, set_attributes=True)
  File "/appl/AS/4.1.14/lib64/python3.9/site-packages/autosubmit/job/job.py", line 2221, in update_parameters
    parameters = self.update_platform_associated_parameters(as_conf, parameters, parameters['CHUNK'], set_attributes)
  File "/appl/AS/4.1.14/lib64/python3.9/site-packages/autosubmit/job/job.py", line 1855, in update_platform_associated_parameters
    self.process_scheduler_parameters(self.platform, chunk)
  File "/appl/AS/4.1.14/lib64/python3.9/site-packages/autosubmit/job/job.py", line 1809, in process_scheduler_parameters
    self.custom_directives = json.loads(self.custom_directives)
  File "/usr/lib64/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 53 (char 52)

[CRITICAL] 2025-10-13 15:19:38,322 Unexpected error: Expecting value: line 1 column 53 (char 52).
 Please report it to Autosubmit Developers through Git
2025-10-13 15:19:38,322 More info at https://autosubmit.readthedocs.io/en/master/troubleshooting/error-codes.html
```
The typo was in line 67, column 81 in the main.yml.

### Reproducible Example

Add a typo in the yaml:
```JOBS:
    TRANSFER:
        PARTITION: standard
        WALLCLOCK: 02:00
        CUSTOM_DIRECTIVES: "['#SBATCH --export=ALL',  #SBATCH --account=xxxxx']"
```
(' missing )
### Expected Behaviour
A meaningful error message could include for which variable the typo was found or which line in the main.yml or which job it affects, depending on what is possible. That way one doesn't need to go through the entire main.yml to find it.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,6 +40,7 @@ platform does not support wrappers anymore (it was used for testing).
 - Added "CUSTOM_" directives support in the wrapper configuration #2669
 - Added a platform option to compress log files of the job's output during log recovery #2555
 - Added automated citation instructions to the website landing page #2480
+- Improving the error message handling for incorrect YAML syntax (too cryptic) #2651
 
 ### 4.1.15: Bug fixes, enhancements, and new features
 

--- a/autosubmit/job/job.py
+++ b/autosubmit/job/job.py
@@ -1843,7 +1843,11 @@ class Job(object):
                 self.custom_directives = self.het['CUSTOM_DIRECTIVES'][0]
             else:
                 if type(self.custom_directives) is str:  # TODO This is a workaround for the time being, just defined for tests passing without more issues
-                    self.custom_directives = json.loads(self.custom_directives)
+                    try:
+                        self.custom_directives = json.loads(self.custom_directives)
+                    except (ValueError, TypeError) as e:
+                        raise AutosubmitCritical(f"Error parsing custom directives: '{self.custom_directives}: {e}'", 6000)
+
             if len(self.het['CUSTOM_DIRECTIVES']) < self.het['HETSIZE']:
                 for x in range(self.het['HETSIZE'] - len(self.het['CUSTOM_DIRECTIVES'])):
                     self.het['CUSTOM_DIRECTIVES'].append(self.custom_directives)
@@ -2649,7 +2653,7 @@ class Job(object):
         return True
 
     def write_vertical_time(
-        self, count: int = -1, first_submit_timestamp: str = None
+        self, count: int = -1, first_submit_timestamp: str = ''
     ) -> None:
         self.update_start_time(count=count)
         self.update_local_logs(update_submit_time=False, count=count)

--- a/test/unit/test_job.py
+++ b/test/unit/test_job.py
@@ -595,6 +595,7 @@ CONFIG:
             self.job.nodes = test['nodes']
             assert self.job.total_processors == test['expected']
 
+
     def test_get_from_total_stats(self):
         """
         test of the function get_from_total_stats validating the file generation
@@ -1983,3 +1984,13 @@ def test_job_parameters_resolves_all_placeholders(autosubmit_config, monkeypatch
     assert parameters["CURRENT_JOB_HAS_PRIO"] == "whatever"
     assert parameters["CURRENT_WRAPPER_HAS_PRIO"] == "whatever_from_wrapper"
     assert parameters["CURRENT_PLATFORM_HAS_PRIO"] == "whatever_from_platform"
+
+
+def test_process_scheduler_parameters(local):
+    job = Job(_EXPID, '1', 'WAITING', 0, None)
+    job.het = {}
+    job.platform = local
+    job.custom_directives = "['#SBATCH --export=ALL',  #SBATCH --account=xxxxx']"
+
+    with pytest.raises(AutosubmitCritical):
+        assert isinstance(job.process_scheduler_parameters(local, 0), AutosubmitCritical)

